internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -92,13 +95,21 @@ func getTemplateData(ctx context.Context, rr *odhtypes.ReconciliationRequest) (m
 		}
 
 		// only when either storage or resources is set, we take replicas into account
-		// - if user did not set it / zero-value "0", we use default value of 2
+		// - if user did not set it / zero-value "0", we choose default based on cluster size (1 for SNO, else 2)
 		// - if user set it to Y, we pass Y to template
-		var replicas int32 = 2 // default value to match monitoringstack CRD's default
-		if (metrics.Storage != nil || metrics.Resources != nil) && metrics.Replicas != 0 {
-			replicas = metrics.Replicas
+		allowedByConfig := metrics.Storage != nil || metrics.Resources != nil
+		isSNO := isSingleNodeCluster(ctx, rr)
+
+		switch {
+		case metrics.Replicas != 0 && allowedByConfig:
+			templateData["Replicas"] = strconv.Itoa(int(metrics.Replicas))
+		case isSNO: // enforce 1 replica on single-node even if Storage/Resources are not set
+			templateData["Replicas"] = "1"
```

Comment:
should this isSNO part of the allowedByConfig == true?


if metrics is set (either stroage or resources or both):
- use what has been set there
- or 1 if SNO, 2 if not SNO

Reply:
yes you're right i'll update the logic